test/files/neg/t7636.scala

```diff
@@ -1,7 +1,7 @@
-object Main extends App{
-  class ResultTable[E]( query : Either[_,E] )( columns : Int )
-  class C extends ResultTable(Left(5):Either[_,_])(5)
+object Main extends App {
+  class Foo[A](x: A)
+  object bar extends Foo(5: T forSome { type T })
 }
-// Inference of the existential type for the parent type argument
-// E still fails. That looks tricky to fix, see the comments in scala/bug#7636.
+// Inference of the existential type for the parent type argument A still fails.
+// That looks tricky to fix, see the comments in scala/bug#7636.
 // But we at least prevent a cascading NPE.
```

Comment:
Any reason for the test change?  I assume the intent is a little more minimal equivalent, which eyeballing it looks like, but is it?  I tend to prefer the user's original, particularly as `Left(5):Either[_,_]` is plausibly user code over `5: T forSome { type T }` - IMHO.

Reply:
The original now compiles because `Either` is covariant and `E` is inferred as `Any` (`ExistentialType.rewrap` just calls `existentialAbstraction` which eliminates simple covariant existentials.